src/components/graph/Graph.jsx

```diff
@@ -328,9 +333,10 @@ export default class Graph extends React.Component {
     onClickNode = clickedNodeId => {
         if (this.state.config.collapsible) {
             const leafConnections = getTargetLeafConnections(clickedNodeId, this.state.links, this.state.config);
-            const links = toggleLinksMatrixConnections(this.state.links, leafConnections, this.state.config);
+            const filteredLeafConnections = filterNonCollapsibleNodes(this.state.nodes, leafConnections);
```

Comment:
I would move this function call to `getTargetLeafConnections`. Making the function private to the module`collapse.helper`.